pkg/kubernetes/pod/v1alpha1/pod.go

```diff
@@ -67,23 +67,36 @@ func (b *listBuilder) List() *podList {
 	return filtered
 }
 
-// Len returns the number of items present in the podList
-func (p *podList) Len() int {
+// Len returns the number of items present in the PodList
+func (p *PodList) Len() int {
 	return len(p.items)
 }
 
-// ToAPIList converts podList to API podList
-func (p *podList) ToAPIList() *v1.PodList {
+// ToAPIList converts PodList to API PodList
+func (p *PodList) ToAPIList() *v1.PodList {
 	plist := &v1.PodList{}
 	for _, pod := range p.items {
 		plist.Items = append(plist.Items, *pod.object)
 	}
 	return plist
 }
 
-// ListBuilder returns a instance of listBuilder
-func ListBuilder() *listBuilder {
-	return &listBuilder{list: &podList{items: []*pod{}}}
+// NewListBuilder returns a instance of ListBuilder
+func NewListBuilder() *ListBuilder {
+	return &ListBuilder{list: &PodList{items: []*pod{}}}
+}
+
+// ListBuilderForAPIList returns a instance of ListBuilder from API PodList
+func ListBuilderForAPIList(pods *v1.PodList) *ListBuilder {
+	b := &ListBuilder{list: &PodList{}}
+	if pods == nil {
+		return b
+	}
+	for _, p := range pods.Items {
+		p := p
+		b.list.items = append(b.list.items, &pod{object: &p})
+	}
+	return b
```

Comment:
Since builder methods & object methods are scattered across this file; move the build methods to `build.go` & `buildlist.go` corresponding to Builder & ListBuilder structures.